dagshub/data_engine/model/datasource.py

```diff
@@ -14,7 +15,7 @@
 from pathlib import Path
 from typing import Any, Dict, List, Optional, TYPE_CHECKING, Union, Set, ContextManager, Tuple, Literal
 
-
+import pandas as pd
```

Comment:
`pandas` is already lazy loaded at line 67, so adding an eager import is redundant.
Get rid of this and swap the calls in `_convert_file_to_df` to use `pandas` instead of `pd`